Tasks/AzureAppServiceManage/appinsightsmanage.ts

```diff
@@ -0,0 +1,163 @@
+                break;
+            }
+        }
+
+        if(isAppInsightsAlreadyConfigured) {
+            tl.debug("App service '" + this.webAppName + "' is already linked to app insights : " + appInsightsResource.name)
+        } else {
+            appInsightsResourceTags[appInsightsLinkingTag] = "Resource";
+            appInsightsResource = await azureRESTUtils.updateApplicationInsightsResource(this.endpoint, this.appInsightsResourceGroupName, appInsightsResource);
+        }
+
+        return appInsightsResource;
+    }
+
+    private async getAppServiceHiddenLink() {
+        var appServiceDetails = await azureRESTUtils.getAppServiceDetails(this.endpoint, this.resourceGroupName, this.webAppName, this.specifySlotFlag, this.slotName);
+        return "hidden-link:" + appServiceDetails.id;
+    }
+
+    private async configureInstrumentationKey(appInsightsResource) {
+        var appInsightsInstrumenationKey = appInsightsResource.properties.InstrumentationKey;
+        var appServiceAppSettings = await azureRESTUtils.getWebAppAppSettings(this.endpoint, this.webAppName, this.resourceGroupName, this.specifySlotFlag, this.slotName);
+
+        if(appInsightsInstrumenationKey && appServiceAppSettings) {
+            appServiceAppSettings.properties["APPINSIGHTS_INSTRUMENTATIONKEY"] = appInsightsInstrumenationKey;
+            appServiceAppSettings = await azureRESTUtils.updateWebAppAppSettings(this.endpoint, this.webAppName, this.resourceGroupName, this.specifySlotFlag, this.slotName, appServiceAppSettings);
+        }
+    }
+
+    private async configureAppServiceAlwaysOnProperty() {
+        try {
+            var appServiceWebSettings = await azureRESTUtils.getAzureRMWebAppConfigDetails(this.endpoint, this.webAppName, this.resourceGroupName, this.specifySlotFlag, this.slotName);
+            if(appServiceWebSettings && appServiceWebSettings.properties && appServiceWebSettings.properties.alwaysOn == false) {
+                    var configSettings = JSON.stringify({
+						"properties": {
+							"alwaysOn" : true
+						}
+					});
+                    appServiceWebSettings = await azureRESTUtils.updateAzureRMWebAppConfigDetails(this.endpoint, this.webAppName, this.resourceGroupName, this.specifySlotFlag, this.slotName, configSettings);
+            }
+        } catch (err) {
+            tl.warning(tl.loc("FailedToConfigureAlwaysOnProperty", JSON.stringify(err)));
+        }
+    }
+
+    private async configureAppInsightsWebTest(appInsightResource, publishingProfile) {
+        try {
+            var allWebTestsInRG = await azureRESTUtils.getAppInsightsWebTests(this.endpoint, this.appInsightsResourceGroupName);
+            var isWebTestAlreadyConfigured = false;
+            
+            for(var webTest of allWebTestsInRG) {
+                var webTestTags = webTest.tags;
+                if(webTestTags) {
+                    for(var tagKey in webTestTags) {
+                        if(tagKey.indexOf(appInsightResource.id) != -1) {
```

Comment:
Can you give a sample of how the tagKey looks.

Reply:
hidden-link:/subscriptions/<SUBSCRIPTIONID>/resourceGroups/abcRG/providers/Microsoft.Web/sites/abcApp